src/app/blog/[title]/_components/GenerateTOC.tsx

```
@@ -0,0 +1,97 @@
+        const aClass = `block pb-1 hover:underline transform transition-transform duration-200 ${selectedIndex === index ? "scale-105 text-gray-900" : ""}`;
+
+        return (
+          <li
+            key={index}
+            className={`${liClass} leading-tight`}
+            role="menuitem"
+            onClick={() => {
+              setSelectedIndex(index);
+              document.getElementById(id)?.scrollIntoView({
+                behavior: "smooth",
+                block: "start",
```

Biome

[error] The HTML element li is non-interactive and should not have an interactive role.

Replace li with a div or a span.
Unsafe fix: Remove the role attribute.


(lint/a11y/noNoninteractiveElementToInteractiveRole)